packages/react/src/numberfield/NumberField.tsx

```diff
@@ -91,14 +101,29 @@ function NumberField(props: NumberFieldProps, ref: Ref<HTMLInputElement>) {
           {leftAddon}
           {withAddonDivider && leftAddon && <InputAddonDivider />}
           <Input
-            className={inputWithAlignment({ textAlign, isGrouped: true })}
+            className={inputVariants({
+              textAlign,
+              isGrouped: true,
+              autoWidth: !!size,
+            })}
             ref={ref}
+            size={
+              size && size + 1
+            } /** Add one extra character for a more precise width */
+            maxLength={maxLength}
           />
           {withAddonDivider && rightAddon && <InputAddonDivider />}
           {rightAddon}
         </Group>
       ) : (
-        <Input className={inputWithAlignment({ textAlign })} ref={ref} />
+        <Input
+          className={inputVariants({ textAlign, autoWidth: !!size })}
+          ref={ref}
+          size={
+            size && size + 1
```

Comment:
Hvorfor pluss 1? Er det fordi man ønsker å ta høyde for innebygget padding i tekstfeltet?

Reply:
Den kommentaren `/** Add one extra character for a more precise width */` ser ut til ha lagt seg under linja her, men er altså for å ha litt sikkerhetsmargin. Men ja, når jeg testet så var det ikke helt plass til nøyaktig like mange karakterer som ble satt med `size`. Godt mulig det er relatert til paddingen på input-feltet her ja.

Reply:
Ja. Jeg er litt usikker på om det er noe vi har lyst til å løse i grunnmuren? Blir det ikke litt rart å sende inn 15, så kommer 16 ut i HTMLen?

Reply:
Er godt mulig det kan bli litt rart, enig i det. Men spørs hva som er mest rart:

1. Du sender inn en `size={15}` men så får det ikke plass 15 tegn.
2. Du sender inn `size={15}`, og hvis du inspecter elementet ser du at den har fått `size={16}`.

Burde for så vidt være mulig å løse dette med CSS. Jeg kan se litt på det

Reply:
Jeg tror ikke vi trenger å være smarte her. Jeg tenker mest på `size` som "circa omtrentlig nogenlunde rundt denne størrelsen".

Selv i plain HTML sizes den ikke helt riktig til akkurat antall tegn du spesifiserer i attributtet.

  
<img width="592" alt="Screenshot 2024-04-12 at 09 59 39" src="https://github.com/code-obos/grunnmuren/assets/81577/88bf7470-53e3-4ec9-a41c-17bdaf065e4a">

Og av erfaring vet vi at designerne foretrekker at feltet faktisk er litt bredere enn antall tillatt tegn også, så man slenger gjerne på litt ekstra size uansett.

Reply:
Ja, jeg kan se den. Jeg tenker også at man i mange tilfeller ønsker litt bredere enn antall tegn. Men jeg gjorde en liten fiks med CSS nå som gjør at det skalerer mer riktig til antall tegn i hvert fall 👍 så fjernet den inkrementeringen - ikke ofte man får bruk for `box-sizing: content-box` 😄